tests/api/api_handler.py

```diff
@@ -0,0 +1,111 @@
+import time
+
+import tests.convenience as conv
+import tests.api.parser as parser
+from tests.api.sender import ApiSender as Sender
+from tests import utils as ut
+
+
+layers_query_request = '{{"start_layer": {{"number": {start_layer}}}, "end_layer": {{"number": {end_layer}}}}}'
+
+
+class ApiHandler:
+    def __init__(self, ips, namespace):
+        self.sender = Sender(ips, namespace)
+
+    def extend_ips(self, ips):
+        self.sender.extend_ips(ips)
+
+    def remove_ips(self, ips):
+        self.sender.remove_ips(ips)
+
+    # TODO: add timeout decorator
+    def wait_for_layer(self, layer_num, timeout=None, interval=10):
+        """
+        wait for layer_num
+        raises ValueError exception if layer has already passed
+
+        :param layer_num: int, wanted layer to wait for
+        :param timeout: int, seconds until timeout
+        :param interval: int, seconds between iterations
+        :return:
+        """
+        is_timeout = timeout is not None
+        if is_timeout and timeout <= 0:
+            raise TimeoutError(f"got a timeout while waiting for layer {layer_num}")
+        current_layer, elapsed_time = self.sender.send_current_layer_timed(timeout)
+        if current_layer > layer_num:
+            # raise a value error in case wanted layer has passed
+            # raise ValueError(f"layer {layer_num} has already passed")
+            return current_layer
+        while True:
+            if current_layer == layer_num:
+                break
+            if is_timeout and timeout <= 0:
+                raise TimeoutError("wait for layer reached timeout")
+            print(f"waiting for layer {layer_num}")
+            print(f"sleeping for {interval} before querying for current layer again\n")
+            time.sleep(interval)
+            timeout = timeout - interval if is_timeout else timeout
```

Comment:
besides my personal preference for the standart `if else` statements you have 2 checks on `if is_timeout` which I think can be refactored if you use this standart

Reply:
checking if 'is_timeout' to see if the timeout is relevant since its default value is None.
first check is for validating we didn't exceed timeout and the second is for recalculating timeout after sleeping for our interval between retries.